### PR TITLE
Make apolloInMemoryCache variables optional

### PR DESCRIPTION
**Pull request by Gregoirevda**

### CODE REVIEW

**Comment by Gregoirevda** [comment]
If someone could review that plz

File: src/ApolloInMemoryCache.re [hunk 1]
```diff
@@ -36,15 +36,22 @@ let createIntrospectionFragmentMatcher = (~data) =>
...
+    | (Some(dataIdFromObject), None) => 
     apolloInMemoryCache({
-      "dataIdFromObject": Js.Nullable.from_opt(dataIdFromObject)
+      "dataIdFromObject": Js.Nullable.return(dataIdFromObject)
     })
+    | (None, Some(fragmentMatcher)) => 
+    apolloInMemoryCache({
+      "fragmentMatcher": Js.Nullable.return(fragmentMatcher)
+    })
+    | (None, None) => apolloInMemoryCache()
```
Comment (grsabreu) [inline src/ApolloInMemoryCache.re]: I think this needs to be `apolloInMemoryCache(Js.Obj.empty())`

**Comment by grsabreu** [comment]
```reason
[@bs.new]
external apolloInMemoryCache : 'a => apolloCache = "InMemoryCache";
```
I think could be

```reason
[@bs.new]
external apolloInMemoryCache : Js.t({..}) => apolloCache = "InMemoryCache";
```

to be more restrictive

**Comment by Gregoirevda** [comment]
Thanks @grsabreu! Will upgrade version and publish

**Comment by grsabreu** [comment]
```reason
[@bs.obj]
  makeApolloInMemoryCacheParams : (~dataIdFromObject: (Js.t({..}) => string)=?, ~fragmentMatcher: (Js.t({..}) => string)=?) => _ = ""
...
    apolloInMemoryCache(
     makeApolloInMemoryCacheParams(~dataIdFromObject=(obj => obj##id), ())
   );
   // in JS { dateIdFromObject: ... }
```

that would also only generate the keys for JS as they are defined

**Comment by Gregoirevda** [comment]
That looks even better

**Comment by Gregoirevda** [comment]
https://github.com/apollographql/reason-apollo/pull/38

**Pull request merged**